bin/analysis.sh

```diff
@@ -17,8 +17,21 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-base_dir=$(dirname "$0")
+# resolve links - $0 may be a softlink (code as used by gradle)
+PRG="$0"
+
+while
+  base_dir=${PRG%"${PRG##*/}"}
+  [ -h "$PRG" ]
+do
+  ls=$( ls -ld "$PRG")
+  link=${ls#*' -> '}
+  case $link in
+    /* )  PRG=$link;;
+    *  )  PRG=$base_dir$link;;
+  esac
+done
+base_dir=$(dirname "$PRG")
```

Comment:
base_dir=$( cd "${base_dir:-./}" && pwd -P ) || exit

Reply:
Thanks for your advice, i've fixed it. Please review it.